.gitignore

```diff
@@ -63,3 +63,5 @@ uImage
 .DS_Store
 tools/gdb/__pycache__
 /build
+.ccls-cache
+compile_commands.json
```

Comment:
Why did you include these file to .gitignore ?

Reply:
`compile_commands.json` is a file that lists the exact compiler calls for each file (that gets compiled). This helps language servers to build their internal AST. CMake [allows it out of the box](https://cmake.org/cmake/help/latest/variable/CMAKE_EXPORT_COMPILE_COMMANDS.html) and so does clang, or it can be created using a CLI tool like `bear` or `compiledb`.

It's a pretty popular file that all C/C++ LSPs use, and even the Linux kernel has a [dedicated script to generate it](https://github.com/torvalds/linux/blob/master/scripts/clang-tools/gen_compile_commands.py).

Since this lists exact compiler calls, it might be different depending on need, branch, state of the code, etc., it is kept in .gitignore.

---

CCLS is one such LSP, which also creates a `.ccls-cache` directory for caching the AST of the repo.